packages/lib-subject-viewers/src/VolumetricViewer/VolumetricViewer.js

```diff
@@ -1,52 +1,56 @@
-import { object, string } from 'prop-types'
+import { func, object, string } from 'prop-types'
 import { useEffect, useState } from 'react'
-import { Buffer } from 'buffer'
 import { ComponentViewer } from './components/ComponentViewer.js'
 import { ModelViewer } from './models/ModelViewer.js'
 import { ModelAnnotations } from './models/ModelAnnotations.js'
 import { ModelTool } from './models/ModelTool.js'
+import { useSubjectJSON } from './../hooks/useSubjectJSON.js'
+import asyncStates from '@zooniverse/async-states'
+
+const DEFAULT_HANDLER = () => {}
 
 export default function VolumetricViewer ({
-  config = {},
-  subjectData = '',
-  subjectUrl = '',
-  models
+  loadingState = asyncStates.initialized,
+  onError = DEFAULT_HANDLER,
+  onReady = DEFAULT_HANDLER,
+  subject
 }) {
-  const [data, setData] = useState(null)
-  if (!models) {
-    const [modelState] = useState({
-      annotations: ModelAnnotations(),
-      tool: ModelTool(),
-      viewer: ModelViewer()
-    })
-    models = modelState
-  }
+  const [subjectData, setSubjectData] = useState({ error: null, data: null })
+  const [modelState] = useState({
+    annotations: ModelAnnotations(),
+    tool: ModelTool(),
+    viewer: ModelViewer()
+  })
 
-  // Figure out subject data
   useEffect(() => {
-    if (subjectData !== '') {
-      setData(Buffer.from(subjectData, 'base64'))
-    } else if (subjectUrl !== '') {
-      fetch(subjectUrl)
-        .then((res) => res.json())
-        .then((data) => {
-          setData(Buffer.from(data, 'base64'))
-        })
-    } else {
-      console.log('No data to display')
-    }
-  }, [])
+    useSubjectJSON({ setSubjectData, subject })
+  }, [subject])
```

Comment:
It's a bit difficult to see from the diff, but this is the pattern you've got here:
```
  useEffect(() => {
    useSubjectJSON({ setSubjectData, subject })
  }, [subject])
```

On first look, this is a custom hook inside `useEffect` which goes against the [React Rules of Hooks](https://react.dev/reference/rules/rules-of-hooks#only-call-hooks-at-the-top-level) "Do not call Hooks inside functions passed to useMemo, useReducer, or useEffect." However, looking closer at your function named `useSubjectJSON()` in lib-subject-viewers, it's not actually a hook.

`useSubjectJSON()` in **lib-classifier** utilizes `useEffect` and `useState` to fetch and return subject data to the viewer. It's a [custom React hook](https://react.dev/learn/reusing-logic-with-custom-hooks#custom-hooks-sharing-logic-between-components) and allowed the "use" naming convention because `useEffect` and `useState` are inside the `useSubjectJSON()` function.

In this PR, you've got `useEffect` and `useState` inside VolumetricViewer.js instead of inside `useSubjectJSON()`, which is just a fetcher function.

Is this pattern intentional?  Let me know if you'd like to have a call to discuss custom hooks. 

Overall, I recommend:
1. Naming your fetcher function in lib-subject-viewers something specific to volumetric data like `useVolumetricSubject()`.
2. Pass `onError`, `onReady`, and `subject` to a custom hook `useVolumetricSubject()` exactly like it's done below for JSONDataViewer. The custom hook `useVolumetricSubject()` then returns `subjectData`, and `error`. This pattern would clean up the code in VolumetricViewer.js, removing the need for two `useEffect` functions, and the need to store `subjectData` and error status in a `useState`.

https://github.com/zooniverse/front-end-monorepo/blob/fbab1d8ef4e43b1f8fe0f43a27bea1c34e49d109/packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/JSONDataViewer/JSONDataViewer.js#L45

Reply:
Got this incorporated as well - I appreciate the focus on the hook's interface within the `VolumetricViewer` component. That helped me refactor/isolate where the state should live.